### ISSUE TITLE
Improve and document "Release" action for feature and patch releases

### ISSUE THREAD

**Issue by morri-son** [open-component-model/ocm-project#240]
**Description**
There should be a standard release process for feature and patch releases of the OCM lib / CLI. In addition to that there should be a nightly build that creates the OCM binaries and a Docker image. The process should look like follows:

* Every four weeks on Friday there is a release decision for a new feature release. On a positive decision
  - a new release is created by triggering the "Release" action in the "ocm" repository on the "main" branch
  - if triggered on the main branch, the release action ALWAYS creates a new release branch as workspace for potential patches
  - if triggered on the main branch, the release action deletes all old release branches, but the new and last 
  - after a successful release the PR that gets automatically created in the ocm-website repo to update the version should be merged.

* On every important bugfix we create a patch for the last released version
  - Cherry-pick the commit for the bugfix from the main branch
  - Trigger "Release" action on latest release branch
  - after a successful release the PR that gets automatically created in the ocm-website repo to update the version should be merged.

* Every night there should be a build of the latest sources from the main branch. The used commit should be tagged with "nightly" and there should be one (pre-created) release also called "nightly". That nightly release should get assigned with the new nightly tag on every successful build. The assets of the release for sure should be the ones created by the last nightly build. Instead of creating a new Docker image with the OCM CLI on every push to the main branch, the nightly build should push that image and tag it with "latest".

**Tasks:**

* Document the process in our repo for internal processes https://github.com/open-component-model/internal-processes.
* Changes to be done in the Github action "Release":
  - remove Option "Create Release Branch" from the action
  - add automatic creation of release branch when action is triggered on branch main
  - add deletion of all old release branches, but the current and last release branch
  - either enhance the action to allow being scheduled for the nightly build or copy and adopt the existing action to a new action "Nightly Build". 

**Done Criteria**
- [x] Code has been reviewed by other team members
- [x] Internal technical Documentation created/updated
- [x] New / changed code is documented
- [x] Successful demonstration in Review

**Comment by hilmarf** [comment]
and check the release notes creation https://github.com/open-component-model/ocm-project/issues/24

**Comment by hilmarf** [comment]
nightly build - https://github.com/open-component-model/ocm/pull/903

**Comment by morri-son** [comment]
document Release process - https://github.com/open-component-model/internal-processes/pull/12

**Comment by morri-son** [comment]
@hilmarf , this issue has been done and can be closed, right?

**Comment by morri-son** [comment]
> [@morri-son](https://github.com/morri-son)
> 
> > this issue has been done and can be closed, right?
> 
> yes and no... I bet, when we do [#284](https://github.com/open-component-model/ocm-project/issues/284) it needs some review and adjustment again. Also from today experience, I'm not sure if it's clear for everyone.

ok, I assigned it as sub-issue to the new Epic that we have to improve the release process.

**Comment by jakobmoellerdev** [comment]
> Document the process in our repo for internal processes https://github.com/open-component-model/internal-processes.

Documented, but now lives in the OCM repo under RELEASE_PROCESS.md

> Changes to be done in the Github action "Release":
> remove Option "Create Release Branch" from the action

Done, now a separate Branch Cutoff action: https://github.com/open-component-model/ocm/blob/main/.github/workflows/release-branch.yaml

> add automatic creation of release branch when action is triggered on branch main

Done, see action above

> add deletion of all old release branches, but the current and last release branch

Not done: Explanation is that we do not have much overhead with retaining old release branches and it is not necessary to be removed. Its useful to have the branches present because the tags (such as v0.17.0) can then be associated with the branch at a given timestamp / commit.

> either enhance the action to allow being scheduled for the nightly build or copy and adopt the existing action to a new action "Nightly Build".

we have a latest build that is currently sufficient but we plan to revisit this topic as soon as we are able to get the component builds used for our latest (currently the latest action is separate). Nevertheless I have not received feedback that this was insufficient yet. 

TLDR: Considering closed, we need to optimize the latest build down the road. I created https://github.com/open-component-model/ocm-project/issues/334 to follow up on this.
